recipes/libsodium/1.0.18/conanfile.py

```diff
@@ -0,0 +1,223 @@
+from conans import ConanFile, AutoToolsBuildEnvironment, tools, MSBuild
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class LibsodiumConan(ConanFile):
+    name        = "libsodium"
+    version     = "1.0.18"
+    description = "A modern and easy-to-use crypto library."
+    license     = "ISC"
+    url         = "https://github.com/conan-io/conan-center-index"
+    homepage    = "https://download.libsodium.org/doc/"
+    exports_sources = ["patches/**"]
+    settings    = "os", "compiler", "arch", "build_type"
+    topics = ("sodium", "libsodium", "encryption", "signature", "hashing")
+    generators  = "cmake"
+    _source_subfolder = "source_subfolder"
+
+    options = {
+        "shared" : [True, False],
+        "fPIC": [True, False],
+        "use_soname" : [True, False],
+        "use_pie"    : [True, False],
+    }
+
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "use_soname": True,
+        "use_pie": False,
+    }
+
+    @property
+    def _android_id_str(self):
+        return "androideabi" if str(self.settings.arch) in ["armv6", "armv7"] else "android"
+
+    @property
+    def _arch_id_str_compiler(self):
+        return {"x86": "i686",
+                "armv6": "arm",
+                "armv7": "arm",
+                "armv7hf": "arm",
+                "armv8": "aarch64",
+                "mips64": "mips64"}.get(str(self.settings.arch),
+                                        str(self.settings.arch))
+
+    @property
+    def _is_mingw(self):
+        return self.settings.os == "Windows" and self.settings.compiler == "gcc"
+
+    @property
+    def _vs_platform(self):
+        if self.settings.arch.value == "x86":
+            return "Win32"
+        return "x64"
+
+    @property
+    def _vs_configuration(self):
+        configuration = ""
+        if self.options.shared:
+            configuration += "Dyn"
+        else:
+            configuration += "Static"
+        build_type = str(self.settings.build_type)
+        if build_type == "RelWithDebInfo":
```

Comment:
`build_type = 'Debug' if self.settings.build_type == 'Debug' else 'Release'`
might be a better condition
cause there are other build types (e.g. `MinSizeRel`), and possibly even custom build types (custom settings.yml)